### ISSUE TITLE
User add 'Never Expires' Option incorrectly setting expiration dates in Chamilo on new accounts

### ISSUE THREAD

**Issue by juancp-contidosdixitais** [chamilo/chamilo-lms#5538]
In a Chamilo site with 'Account validity' configured, if we create a new user with the 'Never expires' option checked, the save process incorrectly sets an expiration date instead of saving it as null. If ''Account validity'' is empty, the process is carried out correctly.

https://github.com/chamilo/chamilo-lms/blob/03e1960b1d83e4df59f1203cdfaa7380d9104ba4/main/admin/user_add.php#L399-L403

https://github.com/chamilo/chamilo-lms/blob/03e1960b1d83e4df59f1203cdfaa7380d9104ba4/main/inc/lib/usermanager.lib.php#L457-L472

If I update an user and check the 'Never expires' option, the save process respects this setting.

Steps to reproduce the problem:
1. Check in Chamilo's settings that 'Account validity' has a value.
2. Create a new user with the 'Never expires' option and save.
3. Verify in the user's record that 'Never expires' is not checked, but an expiration date is set instead.

The expected behavior should be that if the 'Never expires' option is checked, it should be respected even if there is a value set in "Account validity" ... or if "Account validity" has a value, never allow checking the "Never expires" option, I'm not sure which could be the best option ...

- Chamilo version: 1.11.26 (and possibly several earlier versions)"

**Comment by ywarnier** [comment]
Agreed that when "Never expires" is checked, it should behave upon creation as it behaves upon update : set expiration_date to null in the user record.

I've modified the code to do just that. This modifies Chamilo's previous behaviour, but I think it's for the best.
